src/test/unit/types/SdrTest.cpp

```diff
@@ -139,61 +139,48 @@ TEST(SdrTest, TestSDR_Examples) {
 
 TEST(SdrTest, TestSetDenseVec) {
     SDR a({11, 10, 4});
-    Byte *before = a.getDense().data();
-    SDR_dense_t vec = vector<Byte>(440, 1);
-    Byte *data = vec.data();
+    auto before = a.getDense().data();
+    SDR_dense_t vec = SDR_dense_t(440, 1);
+    auto data = vec.data();
     a.setDense( vec );
-    Byte *after = a.getDense().data();
+    auto after = a.getDense().data();
     ASSERT_NE( before, after ); // not a copy.
     ASSERT_EQ( after, data ); // correct data buffer.
 }
 
-TEST(SdrTest, TestSetDenseByte) {
```

Comment:
Where did this test case go?

Reply:
I thought it was not needed due to some casts, but I'll revisit it again! 
Btw, do we need all 3 variants for SDR::setXXX() ? Can we do just with SDR.setDense(SDR_dense_t)?

Reply:
The overloads are:
```
    void setDense( SDR_dense_t &value );
     void setDense( const std::vector<T> &value ) {
     void setDense( const T *value ) {
```
constant & non-constant vectors need different code.
If vector of wrong data type it will do the converstion to Byte.
Raw pointer is for legacy code.

So yes, I think we need all of these.